app/views/games/show.html.erb

```diff
@@ -66,20 +58,33 @@
         var piece = $.parseHTML(
           '<div class="piece" data-id='+ piece.id +'>' + pieceHtml + '</div>');
 
-        $chessboard.find('.square#'+squareId).append(piece);
-        $chessboard.find('.piece').draggable();
+        $('.square#'+squareId).append(piece);
+        $('.piece').draggable();
       }
+    }
 
-      function movePiece(piece) {
-        var coord = piece.closest('.square').attr('id');
-        var newRow = coord[0];
-        var newCol = coord[1];
-
-        $.post(piecesURL + "/" + piece.data("id"), {
-          _method: "PUT",
-          piece: { row: newRow, col: newCol }
-        }).success();
+    $(".square").droppable({
+      drop: function(event, ui) {
+        var coord = $(this).attr('id').split('');
+        var $piece = $(ui.draggable);
+        movePiece($piece, coord);
       }
     });
+
+    // race condition without setTimeout():
+    // The get request may complete before the post request's back-end logic is
+    //   completely executed.
+    // In the worst case, the get request will retrieve pieces before they are
+    //   updated by Piece#move_to!.
```

Comment:
I just figured out what cause the 'race condition'. Previously the code was `.success(setupBoard());`, which invokes the function before the async request is finished. Changing it to `success(setupBoard);` will solve the problem, as I am now passing the function definition, not calling it and passing its return value.